File: survival-client/src/main/java/net/survival/client/input/MouseAdapter.java
package net.survival.client.input;

public abstract class MouseAdapter {
    protected void setMousePosition(double x, double y) {
        Mouse.prevX = Mouse.x;
        Mouse.prevY = Mouse.y;
        Mouse.x = x;
        Mouse.y = y;
    }

    protected void setLmbDown(boolean down) {
        Mouse.prevLmbDown = Mouse.lmbDown;
        Mouse.lmbDown = down;
    }

    protected void setRmbDown(boolean down) {
        Mouse.prevRmbDown = Mouse.rmbDown;
        Mouse.rmbDown = down;
    }

    protected void clearModeChangedFlag() {
        Mouse.modeChanged = false;
    }
}

File: survival-client/src/main/java/net/survival/client/input/MouseAdapter.kt
package net.survival.client.input

abstract class MouseAdapter {
    protected fun setMousePosition(x: Double, y: Double) {
        Mouse.prevX = Mouse.x
        Mouse.prevY = Mouse.y
        Mouse.x = x
        Mouse.y = y
    }

    protected fun setLmbDown(down: Boolean) {
        Mouse.prevLmbDown = Mouse.isLmbDown
        Mouse.isLmbDown = down
    }

    protected fun setRmbDown(down: Boolean) {
        Mouse.prevRmbDown = Mouse.isRmbDown
        Mouse.isRmbDown = down
    }

    protected fun clearModeChangedFlag() {
        Mouse.modeChanged = false
    }
}